### DISCUSSION TITLE
HmIP-DLD "Open" function

### DISCUSSION THREAD

**Discussion by alexanderrock** [SukramJ/aiohomematic#886] [Q&A] [answered]
Hello danielperna,

I've just installed your HomeaticIP Local, great work.
Also installed and connected a HmIP-DLD and I'm lock/unlock is working perfect.
Now I'm searching for the "open" function to open the door.

Maybe I can't find it, or isn't it implemented?

Thanks and Kind Regards

**Comment by Snort007** [comment]
Hi Alexanderrock,
I was also looking about. I have both integrations, the hahomematic and the homematic via configuration yaml.
The HmIP-DLD was only automatically found vie hahomematic but has no open function visible by itself!?
But if I use the mushroom lock card, the open button appears, if the state is unlocked.
If the state is locked, only the unlock button is available.
With the yaml version, I used the work around with the service and the LOCK_TARGET_LEVEL parameter...
Hope this helps!

**Comment by SukramJ** [comment, marked as answer]
The most obvious would be to use the [lock.open](https://www.home-assistant.io/integrations/lock/) service.

**Reply by SukramJ** [reply]
There is also a pending [PR](https://github.com/home-assistant/frontend/pull/14852) to add open support to the frontend.
